lib/components/poster_section.dart

```diff
@@ -24,45 +27,61 @@ const Map<Enum$SectionSize, double> sliderHeight = {
   Enum$SectionSize.medium: 400,
 };
 
-class PosterSection extends StatelessWidget {
+class PosterSection extends ConsumerStatefulWidget {
   final Fragment$Section$$PosterSection data;
 
   const PosterSection(this.data, {super.key});
 
+  @override
+  ConsumerState<PosterSection> createState() => _PosterSectionState();
+}
+
+class _PosterSectionState extends ConsumerState<PosterSection> {
+  Fragment$Episode? curLiveEpisode;
+
   @override
   Widget build(BuildContext context) {
-    if (data.size == Enum$SectionSize.$unknown) {
+    ref.listen<AsyncValue<Fragment$CalendarDayEntries$entries$$EpisodeCalendarEntry?>>(currentLiveEpisodeProvider, (prevEntry, curEntry) {
```

Comment:
You can use `ref.watch` instead of listening. That means we can keep these stateless too 🎉

Reply:
If there is any issue here with it not updating properly (idk how futureprovider works..), you can create a normal provider (in addition?) that returns a simple episodecalendarentry without any future stuff

Reply:
> You can use `ref.watch` instead of listening. That means we can keep these stateless too 🎉

The problem I was facing with `ref.watch` is that the widget was getting rebuilt frequently i.e every 5 seconds regardless of whether the current live episode has changed or not.

Reply:
It should be able to tweak this to use a normal provider or stateprovider that returns a simple episodecalendarentry without any Future or Stream. I think the problem lies there somewhere.

Reply:
> It should be able to tweak this to use a normal provider or stateprovider that returns a simple episodecalendarentry without any Future or Stream. I think the problem lies there somewhere.

Right 👍